### PR TITLE
Open table methods

### PR DESCRIPTION
**Pull request by fergusdixon**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | no
| Fixed issues      | #

## The problem
OpenTable was missing methods
...

## How I resolved it
Added them
...

## Questions
...

Notify the following people: @darrynten

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,13 +65,14 @@ Checked off bits are complete.
...
-  - [x] Items
-  - [ ] Open Table List
-  - [ ] Open Table Detail
-- [x] Vendor Payments
-  - [x] Vendor Payment Collection
+  - [x] Items Collection
+  - [x] Open Table List
+  - [x] Open Table Detail
+- [ ] Vendor Payments
+  - [ ] Vendor Payment Collection
```
Comment (darrynten) [inline README.md]: Oops I think this was me and my merge fix :/ sorry! Please correct :)

File: src/Models/Item.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
  */
 class Item extends BaseModel
 {
+    protected $endpoint = '';
```
Comment (darrynten) [inline src/Models/Item.php]: I think this can be removed from this model completely

File: src/Models/OpenTable.php [hunk 3]
```diff
@@ -82,4 +83,26 @@ class OpenTable extends BaseModel
...
+        return $model;
+    }
+
+    public function list()
+    {
+        $results = $this->request->request('GET', $this->endpoint, 'List');
+        return new ModelCollection(OpenTable::class, $this->config, $results);
+    }
+
+
```
Comment (darrynten) [inline src/Models/OpenTable.php]: Too many newlines

File: tests/PilotLive/Models/BaseModelTest.php [hunk 4]
```diff
@@ -114,15 +96,12 @@ protected function verifySetReadOnly(string $class, string $field)
     protected function verifyGetUndefined(string $class)
     {
         $className = $this->getClassName($class);
-
```
Comment (darrynten) [inline tests/PilotLive/Models/BaseModelTest.php]: Why all these removals?

File: tests/PilotLive/Models/OpenTableModelTest.php [hunk 5]
```diff
@@ -68,4 +68,65 @@ public function testAttributes()
             ],
         ]);
     }
+
+    public function testDetail(){
```
Comment (darrynten) [inline tests/PilotLive/Models/OpenTableModelTest.php]: Space between `()` and `{`

File: tests/PilotLive/Models/OpenTableModelTest.php [hunk 6]
```diff
@@ -68,4 +68,65 @@ public function testAttributes()
...
+        $params = [
+            'reference' => '123abc']
+        ;
```
Comment (darrynten) [inline tests/PilotLive/Models/OpenTableModelTest.php]: No semi colon on own line - the `]` from the line above should be here (and a trailing comma)

**Review by darrynten — CHANGES_REQUESTED** [review]
Some minor change requests

File: tests/PilotLive/Models/BaseModelTest.php (hunk 4 above)
Comment (fergusdixon) [reply tests/PilotLive/Models/BaseModelTest.php]: I ran PHPStorm's code cleanup tool, must not like spaces between methods. Should I change it back?

Comment (darrynten) [reply tests/PilotLive/Models/BaseModelTest.php]: Yep, it's important to use newlines to group logic chunks of functionality in a method and make for lower cognitive load on developers when working on the code in the future.

Always remember that we write code for humans 😎

File: tests/PilotLive/Models/ItemModelTest.php [hunk 7]
```diff
@@ -23,21 +23,29 @@ public function testAttributes()
...
                 'readonly' => false,
+                'min' => 0,
+                'max' => 1000000,
             ],
             'qty' => [
                 'type' => 'double',
                 'nullable' => false,
                 'readonly' => false,
+                'min' => 0,
+                'max' => 1000000,
```
Comment (darrynten) [inline tests/PilotLive/Models/ItemModelTest.php]: Wondering if these should include (and require) the `.00`

File: tests/PilotLive/Models/OpenTableModelTest.php [hunk 8]
```diff
@@ -68,4 +68,68 @@ public function testAttributes()
             ],
         ]);
     }
+
+    public function testDetail()
+    {
+        $params = [
+            'reference' => '123abc',
+            ];
```
Comment (darrynten) [inline tests/PilotLive/Models/OpenTableModelTest.php]: Incorrect indentation

**Review by darrynten — CHANGES_REQUESTED** [review]
Some minor change requests

File: tests/PilotLive/Models/BaseModelTest.php [hunk 9]
```diff
@@ -132,11 +124,9 @@ protected function verifyGetUndefined(string $class)
     protected function verifyCanNotNullify(string $class, string $key)
     {
         $className = $this->getClassName($class);
-
         $this->expectException(ModelException::class);
         $this->expectExceptionMessage("Model \"{$className}\" attempting to nullify key {$key} Property is null without nullable permission");
         $this->expectExceptionCode(10111);
-
         $model = new $class($this->config);
         $model->{$key} = null;
```
Comment (darrynten) [inline tests/PilotLive/Models/BaseModelTest.php]: I think the previous way with the 3 distinct segments, (1) we get the class, (2) we set expectations, (3) we so the thing, is the way to leave it.

Code must tell a story, and stories have paragraphs 😺

File: tests/PilotLive/Models/BaseModelTest.php [hunk 10]
```diff
@@ -199,9 +185,7 @@ protected function verifyAttributes(string $class, array $attributes)
         $reflectValue = $reflect->getProperty('fields');
         $reflectValue->setAccessible(true);
         $value = $reflectValue->getValue(new $class($this->config));
-
         $fieldsCount = count($attributes);
-
```
Comment (darrynten) [inline tests/PilotLive/Models/BaseModelTest.php:204]: This is an example of a good removal

File: tests/PilotLive/Models/BaseModelTest.php [hunk 11]
```diff
@@ -165,13 +154,10 @@ protected function verifyCanNullify(string $class, string $key)
     protected function verifyBadImport(string $class, string $key)
     {
         $className = $this->getClassName($class);
-
```
Comment (darrynten) [inline tests/PilotLive/Models/BaseModelTest.php:168]: Good removal

File: tests/PilotLive/Models/BaseModelTest.php [hunk 12]
```diff
@@ -165,13 +154,10 @@ protected function verifyCanNullify(string $class, string $key)
...
         $model = new $class($this->config);
-
```
Comment (darrynten) [inline tests/PilotLive/Models/BaseModelTest.php]: Not so good removal

**Review by darrynten — APPROVED** [review]
Looking good!

**Pull request merged**